### PR TITLE
Index clean

### PR DESCRIPTION
**Pull request by dinosaure**
It's the final implementation of the INDEX file

### CODE REVIEW

File: src/git-unix/ogit-cat-file/main.ml [hunk 1]
```diff
@@ -77,29 +77,29 @@ let setup_logs style_renderer level =
   quiet, Fmt.stdout
 
 type error =
-  [ `Store of Git_unix.FS.error
-  | `Invalid_hash of Git_unix.FS.Hash.t ]
+  [ `Store of Git_unix.Store.error
```
Comment (samoht) [inline src/git-unix/ogit-cat-file/main.ml]: Maybe a better style to `open Git_unix` at the beginning of the file here.

File: src/git-unix/git_unix.ml [hunk 2]
```diff
@@ -13,13 +13,20 @@
...
+module Net   = Net
+module SHA1  = Git.Hash.Make(Digestif.SHA1)
+module Sync  = Git.Sync.Make(Net)
+module Http  = Http.Make
+module Index = Index
+
+module Store = struct
+  include Git.Store.Make(SHA1)(Fs)(Git.Inflate)(Git.Deflate)
+
+  let with_fs ?root ?dotgit ?compression ?buffer fs =
```
Comment (samoht) [inline src/git-unix/git_unix.ml]: Could you merge `with_fs` with `create` (and call the new function `v`) and have `?(fs = Fs.v ?temp_dir ())` ?

**Comment by samoht** [comment]
Looks good. I will investigate why windows tests are broken.

File: src/git-unix/git_unix.ml [hunk 3]
```diff
@@ -24,10 +24,6 @@ module Index = Index
 module Store = struct
   include Git.Store.Make(SHA1)(Fs)(Git.Inflate)(Git.Deflate)
 
-  let with_fs ?root ?dotgit ?compression ?buffer fs =
-    create ?root ?dotgit ?compression ?buffer fs
-
-  let create ?temp_dir ?root ?dotgit ?compression ?buffer () =
-    let fs = Fs.v ?temp_dir () in
+  let v ?root ?dotgit ?compression ?buffer ?(fs = Fs.v ()) () =
```
Comment (samoht) [inline src/git-unix/git_unix.ml]: you still want to apply `?temp_dir` to `Fs.v`

**Comment by dinosaure** [comment]
Lol githuh missed `appveyor` tests and tell me all is ok, GOING TO MERGE!

**Comment by samoht** [comment]
All green, merging! :-)

**Pull request merged**